src/planner/constify_now.c

```diff
@@ -96,11 +96,11 @@ is_valid_now_expr(OpExpr *op, List *rtable)
 
 	Interval *offset = DatumGetIntervalP(c->constvalue);
 	/*
-	 * We don't consider day or month intervals safe here as
+	 * We don't consider month intervals safe here as
```

Comment:
I feel like we don't consider month intervals here because they are affected by February, not timezones. But I dunno.

Reply:
While the reason we do this is indeed because of month boundaries it's not limited to february. Depending on timezone an UTC value might end up either in the beginning of a month or the end of the previous month which will alter the outcome of any `+ '1 month'` calculation since months dont have static amount of days.

Reply:
I'll do a followup PR where I enable month too just with a bigger safety buffer due to the month boundary thing.